commands/management/add_external_serverless.go

```diff
@@ -0,0 +1,198 @@
+
+	Environment    string
+	Cluster        string
+	ReplicationSet string
+	CustomLabels   string
+	Group          string
+
+	MachineID     string
+	Distro        string
+	ContainerID   string
+	ContainerName string
+	NodeModel     string
+	Region        string
+	Az            string
+}
+
+func (cmd *addExternalServerlessCommand) Run() (commands.Result, error) {
+	customLabels, err := commands.ParseCustomLabels(cmd.CustomLabels)
+	if err != nil {
+		return nil, err
+	}
+
+	scheme, metricsPath, address, port, err := cmd.processURLFlags()
+	if err != nil {
+		return nil, err
+	}
+
+	params := &external.AddExternalParams{
+		Body: external.AddExternalBody{
+			AddNode: &external.AddExternalParamsBodyAddNode{
+				NodeType:      pointer.ToString(external.AddExternalParamsBodyAddNodeNodeTypeREMOTENODE),
+				NodeName:      cmd.Name,
+				MachineID:     cmd.MachineID,
+				Distro:        cmd.Distro,
+				ContainerID:   cmd.ContainerID,
+				ContainerName: cmd.ContainerName,
+				NodeModel:     cmd.NodeModel,
+				Region:        cmd.Region,
+				Az:            cmd.Az,
+				CustomLabels:  customLabels,
+			},
+			Address:        address,
+			ServiceName:    cmd.Name,
+			Username:       cmd.Username,
+			Password:       cmd.Password,
+			Scheme:         scheme,
+			MetricsPath:    metricsPath,
+			ListenPort:     int64(port),
+			Environment:    cmd.Environment,
+			Cluster:        cmd.Cluster,
+			ReplicationSet: cmd.ReplicationSet,
+			CustomLabels:   customLabels,
+			MetricsMode:    pointer.ToString(external.AddExternalBodyMetricsModePULL),
+			Group:          cmd.Group,
+		},
+		Context: commands.Ctx,
+	}
+	resp, err := client.Default.External.AddExternal(params)
+	if err != nil {
+		return nil, err
+	}
+
+	return &addExternalServerlessResult{
+		Service: resp.Payload.Service,
+	}, nil
+}
+
+func (cmd *addExternalServerlessCommand) processURLFlags() (scheme, metricsPath, address string, port uint16, err error) {
+	scheme = cmd.Scheme
+	address = cmd.Host
+	port = cmd.ListenPort
+	metricsPath = cmd.MetricsPath
+
+	switch {
+	case cmd.URL != "":
+		uri, err := url.Parse(cmd.URL)
+		if err != nil {
+			return "", "", "", 0, fmt.Errorf("couldn't parse URL %s : %s", cmd.URL, err)
+		}
+		scheme = uri.Scheme
+		address = uri.Hostname()
+		portS := uri.Port()
+		if portS != "" {
+			portI, err := strconv.Atoi(portS)
+			if err != nil {
+				return "", "", "", 0, err
+			}
+			port = uint16(portI)
+		}
+		metricsPath = uri.Path
+	case cmd.Address != "":
+		host, portS, err := net.SplitHostPort(cmd.Address)
+		if err != nil {
+			return "", "", "", 0, err
+		}
+		address = host
+		portI, err := strconv.Atoi(portS)
+		if err != nil {
+			return "", "", "", 0, err
+		}
+		port = uint16(portI)
+	}
+
+	return scheme, metricsPath, address, port, nil
+}
+
+// register command
+var (
+	AddExternalServerless  = new(addExternalServerlessCommand)
+	AddExternalServerlessC = AddC.Command("external-serverless", "Add External Service on Remote node to monitoring")
```

Comment:
Again, S3 is more than "Service on Remote node".
Using the serverless keyword can lead to misunderstanding and future braking changes.